src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -219,7 +220,11 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                 {
                     var result = await ptr(this, context.IncrementStackDepthCounter(childContext), node.IRContext, args);
 
-                    Contract.Assert(result.IRContext.ResultType == node.IRContext.ResultType || result is ErrorValue || result.IRContext.ResultType is BlankType);
+                    if (IfFunction.CanCheckIfReturn(func))
```

Comment:
I proposed Anderson not to filter on some give function or set of functions (If and Switch now) but use the return type instead (which is DType.Unknown)

@CarlosFigueiraMSFT Do you think this is a valid approach?

Reply:
What you think of the following approach? It doesn't seem to break any test.
```
if (!func.ReturnType.IsUnknown)
{
    ...
}
```

Reply:
func.ReturnType is set statically  (see TexlFunction) and the same for all invocations of the function. This is meaningless in cases where the function's type depends on its args. In these cases, some functions set to Unknown (IfFunction does).  

But there are many times when TexlFunction.ReturnType is unknown, but we should still run this assert. 

That's because this assert is checking the binding type (computed once we know the parameter types) against the actual runtime type.  That should always be true, except in very rare cases where the binder is wrong. IfFunction() hits one of those rare cases. 


So the question I'd have is how many functions have this true? Is it just IfFunction()? or others?

Reply:
Kept the original push approach

Reply:
Answering Mike's question: other functions have the same issue (IfError currently does, Switch doesn't but it should)